### DISCUSSION TITLE
How can I create or download larger input files for the examples?

### DISCUSSION THREAD

**Discussion by ztdepztdep** [ginkgo-project/ginkgo#1005] [Q&A] [answered]
I want to test the program in the example directory with a large  A, but how to download it with the command.

**Comment by upsj** [comment, marked as answer]
You can either download a matrix from the [SuiteSparse Matrix Collection](https://sparse.tamu.edu/) in MatrixMarket format, unpack it and use it to replace the `A.mtx` file, or generate one yourself, for example using MATLAB and the `mmwrite` function, whose source code you can find [here](https://math.nist.gov/MatrixMarket/mmio/matlab/mmiomatlab.html).

If you want to download a larger number of matrices, or automate some processes involving downloads from the SuiteSparse Matrix Collection, take a look at the [`ssget`](https://github.com/ginkgo-project/ssget) script.

And again, please use the [Github Discussions](https://github.com/ginkgo-project/ginkgo/discussions/categories/q-a) to ask your questions and avoid creating new issues, unless they involve compilation or execution failures!
